### PR TITLE
Make printer configurable

### LINKED ISSUE #3: Allow to pass a different printer
I would like to build a flat list of unused step definitions and write them in a plain text files for later analysis.

### PR DESCRIPTION
**Pull request by claudiu-cristea**
Fixes #3

### CODE REVIEW

**Comment by nicwortel** [comment]
Hi, thank you for this well-written PR :+1: 

I see that PHPStan fails on the new lines added to `Extension.php`. I have fixed this in the master branch by adding the [phpstan/phpstan-symfony extension](https://github.com/phpstan/phpstan-symfony), but I believe you have to rebase your work on the latest master to trigger a build which includes this fix.

**Comment by claudiu-cristea** [comment]
> (...) but I believe you have to rebase your work on the latest master to trigger a build which includes this fix.

@nicwortel thank you, I did on both PRs but the tests are not running. Needs approval again

File: src/Extension.php [hunk 1]
```diff
@@ -43,7 +46,7 @@ public function load(ContainerBuilder $container, array $config): void
             [
                 new Reference(DefinitionExtension::FINDER_ID),
                 new Reference(DefinitionExtension::REPOSITORY_ID),
-                new Reference('unused_step_definitions_printer'),
+                new Reference($config['printer']),
```
Comment (nicwortel) [inline src/Extension.php:49]: I believe adding a type cast should fix the error from PHPStan:
 
```suggestion
                new Reference((string) $config['printer']),
```

**Comment by nicwortel** [comment]
Thank you! There are some small remaining issues, but I will fix them after merging.

**Pull request merged**